src/mantarray_file_manager/file_writer.py

```diff
@@ -156,3 +165,159 @@ def migrate_to_latest_version(
+        reference_data_start_index >= reference_data_last_index
+        or tissue_data_start_index >= tissue_data_last_index
+    ):
+        raise TooTrimmedError(from_start, from_end, total_time)
+
+    # old metadata
+    old_h5_file = old_file.get_h5_file()
+    old_metadata_keys = set(old_h5_file.attrs.keys())
+    old_from_end = 0
+    old_from_start = 0
+    is_untrimmed = old_h5_file.attrs[str(IS_FILE_ORIGINAL_UNTRIMMED_UUID)]
+
+    if not is_untrimmed:
+        old_from_start = old_h5_file.attrs[str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID)]
+        old_from_end = old_h5_file.attrs[str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID)]
+
+        old_metadata_keys.remove(str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID))
+        old_metadata_keys.remove(str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID))
+        old_metadata_keys.remove(str(IS_FILE_ORIGINAL_UNTRIMMED_UUID))
+
+        old_file_basename = old_file_basename.split("__trimmed")[0]
+
+    if actual_start_trimmed != from_start:
+
+        print(  # allow-print
+            f"{actual_start_trimmed} centimilliseconds was trimmed from the start instead of {from_start}"
+        )
+
+    if actual_end_trimmed != from_end:
+
+        print(  # allow-print
+            f"{actual_end_trimmed} centimilliseconds was trimmed from the end instead of {from_end}"
+        )
+
+    # create new file
+    new_file_name = os.path.join(
+        working_directory,
+        f"{old_file_basename}__trimmed_{actual_start_trimmed + old_from_start}_{actual_end_trimmed + old_from_end}.h5",
+    )
+
+    new_file = MantarrayH5FileCreator(new_file_name)
+
+    for iter_metadata_key in old_metadata_keys:
+        new_file.attrs[iter_metadata_key] = old_h5_file.attrs[iter_metadata_key]
+
+    # new metadata
+    metadata_to_create: Tuple[Tuple[uuid.UUID, Union[str, bool, int, float]], ...]
+
+    metadata_to_create = (
+        (IS_FILE_ORIGINAL_UNTRIMMED_UUID, False),
+        (TRIMMED_TIME_FROM_ORIGINAL_START_UUID, actual_start_trimmed + old_from_start),
+        (TRIMMED_TIME_FROM_ORIGINAL_END_UUID, actual_end_trimmed + old_from_end),
+    )
+
+    for iter_metadata_key, iter_metadata_value in metadata_to_create:
+        new_file.attrs[str(iter_metadata_key)] = iter_metadata_value
+
+    # adding new trimmed data
+    new_tissue_sensor_data = old_h5_file["tissue_sensor_readings"]
+    new_tissue_sensor_data = np.array(new_tissue_sensor_data)
+    new_tissue_sensor_data = new_tissue_sensor_data[
+        tissue_data_start_index : tissue_data_last_index + 1
+    ]  # +1 because needs to be inclusive of last index
+
+    new_reference_sensor_data = old_h5_file["reference_sensor_readings"]
+    new_reference_sensor_data = np.array(new_reference_sensor_data)
+    new_reference_sensor_data = new_reference_sensor_data[
+        reference_data_start_index : reference_data_last_index + 1
+    ]  # +1 because needs to be indclusive of last index
+
+    new_file.create_dataset("tissue_sensor_readings", data=new_tissue_sensor_data)
```

Comment:
note for future.  if these (`"tissue_sensor_readings"` and `"reference_sensor_readings"`) aren't already constants, they could be made so, as I think they're used several places throughout the library